### ISSUE TITLE
Create variable substitution as a discrete pipeline task

### ISSUE THREAD

**Issue by roryprimrose** [Microsoft/azure-pipelines-tasks#8428]
This is a feature request.

## Environment
- Server - Azure Pipelines and local TFS   

- Agent - Hosted and Private

## Description

The web publishing tasks contain options for running variable substitution. For example, pushing pipeline variables down into an appsettings.json file. This is really powerful but only if you are doing a web application publish. Any other application deployment type does not have this substitution power.

Is it possible to release a discrete variable substitution pipeline task so that other deployment mechanics can still use this feature?

**Comment by kmkumaran** [comment]
Yes, This is in our plans as well, but we haven't scheduled this work yet. Adding @RoopeshNair to take note of the request and track.

**Comment by RoopeshNair** [comment]
Thought we have them on our backlog, it will be a while before we can get to it. There are few marketplace extensions that offer tokenization capabilities, my recommendation would be to use them.

**Comment by johnkors** [comment]
@RoopeshNair , could you list them? We're also interested in doing this.

**Comment by RoopeshNair** [comment]
@N-Usha

**Comment by ArieHein** [comment]
@johnkors any of these will do, or just a powershell script that does RegEx

https://marketplace.visualstudio.com/search?term=token&target=AzureDevOps&category=All%20categories&sortBy=Relevance
